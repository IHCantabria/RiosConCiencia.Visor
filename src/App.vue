<template>
  <div id="app" class="app-style grid">
    <viewer class="main"></viewer>
  </div>
</template>
<script>
import Viewer from "@/views/Viewer.vue";

export default {
  components: {
    viewer: Viewer
  },
  mounted() {
    this.init();
  },
  methods: {
    init() {
      this.$spinner.configureModeSpinner(
        this.$spinnerEnums.ModesSpinner.progress
      );
    }
  }
};
</script>
