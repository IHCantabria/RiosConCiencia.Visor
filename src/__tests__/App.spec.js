import { shallowMount } from "@vue/test-utils";
import App from "../App.vue";
import merge from "lodash.merge";

function createWrapper(overrides) {
  const defaultMountingOptions = {
    mocks: {
      $spinner: {
        configureModeSpinner: jest.fn(),
        spinnerConfig: {
          mode: 1
        }
      },
      $spinnerEnums: {
        ModesSpinner: 1
      }
    }
  };
  return shallowMount(App, merge(defaultMountingOptions, overrides));
}

describe("app.vue", () => {
  //UNIT TESTS
  it("spinner mode set after init", () => {
    expect.assertions(2);
    const wrapper = createWrapper();
    const spy = jest.spyOn(wrapper.vm, "init");
    const spySpinner = jest.spyOn(wrapper.vm.$spinner, "configureModeSpinner");
    wrapper.vm.init();
    expect(spy).toHaveBeenCalled();
    expect(spySpinner).toHaveBeenCalled();
  });
  it("init method called after mounted", () => {
    expect.assertions(1);
    const mockInit = jest.fn();
    createWrapper({
      methods: {
        init: mockInit
      }
    });
    expect(mockInit).toHaveBeenCalled();
  });
  //SNAPSHOT TESTS
  it("renders correctly", () => {
    expect.assertions(1);
    const wrapper = createWrapper();
    expect(wrapper.element).toMatchSnapshot();
  });
});
