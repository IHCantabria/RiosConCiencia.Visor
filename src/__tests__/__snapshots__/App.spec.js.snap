// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app.vue renders correctly 1`] = `
<div
  class="app-style grid"
  id="app"
>
  <viewer-stub
    class="main"
  />
</div>
`;
